spec/sec-regexp-regular-expression-objects-patch.html

```diff
@@ -246,181 +284,231 @@ <h1>GetStringIndex ( _S_, _Input_, _e_ )</h1>
       1. Return _eUTF_.
       </emu-alg>
     </emu-clause>
+
+    <emu-clause id="sec-getmatchresult" aoid="GetMatchResult">
+      <h1><ins>GetMatchResult ( _Input_, _startIndex_, _endIndex_, _fullUnicode_ _matchOptions_ )</ins></h1>
+      <p>The abstract operation GetMatchResult with arguments _Input_, _startIndex_, _endIndex_, _fullUnicode_, and MatchOptions _matchOptions_ performs the following steps:</p>
+      <emu-alg>
+      1. Assert: _Input_ is a List of either Unicode code points or code units.
+      1. If _matchOptions_.[[Captures]] is `"indices"`, then
+        1. Return CreateArrayFromList(&laquo; _startIndex_, _endIndex_ &raquo;).
```

Comment:
any reason not to also return the string? Is there a use case for the indices where you don’t end up needing the string?

Reply:
It depends on your use case. Part of the reason that we even need the `options` argument were concerns over memory overhead for adding the offsets on every result. This proposal is attempting to provide the minimal support necessary without additional overhead. It would be easy enough to do this:

```js
const match = regexp.exec(input, { captures: "indices" })
if (match) {
  const substr = match.input.slice(match[1][0], match[1][1]); // substring is recoverable
  ...
}
```

Reply:
Or just `match.input.slice(...match[1])`...

Reply:
sure, but is there any case where someone wants the indices, but won't end up slicing the string? if so it seems much more performant for the engine to do it than for the user to.

Reply:
If it were something that could be lazy initialized, possibly, but there are cases such as with parsing TextMate grammars where you don't need the actual text, just the position information so that you know where to inject syntax highlighting into an existing document.

Reply:
Plus we can always introduce a `{ captures: "indices+strings" }` mode or something similar in the future, or a `{ captures: (start, end, text, group) => { ... } }` callback, given the flexibility of using an `options` object here.